### PR TITLE
https://next-login.dev.buffer.com/

### PR DESCRIPTION
**Pull request by philippemiguet**
Next feature branch for buffer-login, it contains:
 - https://github.com/bufferapp/buffer-login/pull/9
 - https://github.com/bufferapp/buffer-login/pull/13

JIRA: 
- https://buffer.atlassian.net/browse/ACC-201
- https://buffer.atlassian.net/browse/ACC-202

### CODE REVIEW

File: src/login.js [hunk 1]
```diff
@@ -40,37 +24,59 @@ module.exports = async (req, res, next) => {
...
+    const jwt = objectPath.get(session, [PRODUCT_REPLY, 'jwt']);
+    if (app === PRODUCT_REPLY && jwt) {
+      // we have a token, just redirect to Reply
+      return redirectWithParams({
+        baseRoute: redirect || '/',
+        res,
+      })
+    }
+
+    const accessToken = objectPath.get(session, [PRODUCT_PUBLISH, 'accessToken'])
```
Comment (philippemiguet) [inline src/login.js]: @esclapes - Just realized what we did here is not right. APP can be both 'publish' or 'analyze' at this point. If a user tries to log in to Analyze APP will have as a value 'analyze'. Behind the scene, we will run the same logic as for Publish, but the access_token will be saved under the 'analyze' key in the session. That's why Harrison was looping over all the keys.

FYI, in the future Analyze may move away from relying on Publish access token and API and may have their own token.

We need to change both the `objectPath.get` logic and the two if statements below to accept both Publish and Analyze as an APP.

Comment (philippemiguet) [reply src/login.js]: This is how Analyze uses the session: https://github.com/bufferapp/buffer-analyze/blob/450ca19baf4d267dbacf0b2d8cbdf0481def7e8d/packages/server/rpc/user/index.js

**Comment by esclapes** [comment]
@ssvmvss @philippemiguet I just merged here the security update PR and also merged `master` back into `next` to resolve some conflicts. You may want to update your WIP branches that originated from `next`

File: src/app.js [hunk 2]
```diff
@@ -3,6 +3,11 @@ const express = require('express');
 const hbs = require('express-hbs');
 const bugsnag = require('bugsnag');
 const cookieParser = require('cookie-parser');
+const {
+  IS_PRODUCTION,
+  USE_PRODUCTION_SERVICES,
+  SHOULD_USE_BUGSNAG
+} = require('./constants')
```
Comment (philippemiguet) [inline src/app.js]: ❤️

Comment (philippemiguet) [reply src/app.js]: I think that's you @esclapes ^ :D

Comment (esclapes) [reply src/app.js]: :see_no_evil: you spotted me

**Pull request merged**